lib/private/legacy/OC_Util.php

```diff
@@ -313,11 +302,43 @@ public static function setupFS($user = '') {
 			$mountManager->addMount($rootMountProvider);
 		}
 
-		if ($user != '' && !\OC::$server->getUserManager()->userExists($user)) {
+		\OC::$server->getEventLogger()->end('setup_root_fs');
+
+		return true;
+	}
+
+	/**
+	 * Can be set up
+	 *
+	 * @param string $user
+	 * @return boolean
+	 * @description configure the initial filesystem based on the configuration
+	 * @suppress PhanDeprecatedFunction
+	 * @suppress PhanAccessMethodInternal
+	 */
+	public static function setupFS($user = '') {
+		// If we are not forced to load a specific user we load the one that is logged in
+		if ($user === null) {
+			$user = '';
+		} elseif ($user == "" && \OC::$server->getUserSession()->isLoggedIn()) {
+			$user = OC_User::getUser();
+		}
+
+		self::setupRootFS($user);
```

Comment:
Since `setupRootFS` is typehinting string, the only other option is null from this code?
```suggestion
	 * @param string|null $user
	 * @return boolean
	 * @description configure the initial filesystem based on the configuration
	 * @suppress PhanDeprecatedFunction
	 * @suppress PhanAccessMethodInternal
	 */
	public static function setupFS(?string $user = '') {
		// If we are not forced to load a specific user we load the one that is logged in
		if ($user === null) {
			$user = '';
		} elseif ($user === '' && \OC::$server->getUserSession()->isLoggedIn()) {
			$user = OC_User::getUser();
		}

		self::setupRootFS($user);
```

Should make psalm more happy